### PR TITLE
[hl7] Added definitions

### PR DESCRIPTION
**Pull request by MinaroShikuchi**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/hl7/tslint.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "extends": "@definitelytyped/dtslint/dt.json",
+    "rules": {
+        "npm-naming": false,
+        "no-useless-files": false
```
Comment (orta) [inline types/hl7/tslint.json:5]: I'm a little unsure about why these are setting (given that it seems to match an existing npm module?)

Comment (MinaroShikuchi) [reply types/hl7/tslint.json:5]: This package is standalone, it types HL7 Standard version 2.5 which is an annotation like CSV or JSON widely used in Healthcare, you can find more info [here](https://www.hl7.org/)

File: types/hl7/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// Type definitions for non-npm package hl7 2.5
+// Project: https://github.com/MinaroShikuchi/hl7
```
Comment (peterblazejewicz) [inline types/hl7/index.d.ts:2]: this is dead-end url

**Comment by peterblazejewicz** [comment]
`types/hl7/hl7-tests.ts` this can't empty

File: types/hl7/index.d.ts (hunk 2 above)
Comment (MinaroShikuchi) [reply types/hl7/index.d.ts:2]: Fixed

**Comment by MinaroShikuchi** [comment]
> `types/hl7/hl7-tests.ts` this can't empty

Ok so what I'm suppose to test ?

**Comment by peterblazejewicz** [comment]
> Ok so what I'm suppose to test ?

Some real usage would be just enough. This will show us how those are to be used, thx!

**Comment by MinaroShikuchi** [comment]
Something like this ?

```ts
import hl7 from 'hl7';

// Types your Event like this
function handleORM(message: hl7.TriggerEvent['ORM']) {
    // Then it will types your Segment
    const firstname = message.PID['PID.5']['PID.5.1']; // firstname type is string
    const lastname = message.PID['PID.5']['PID.5.1']; // firstname type is string
}
```
It will also give you information about the the field 
image

**Comment by elibarzilay** [comment]
@MinaroShikuchi ping re the above comments

**Comment by peterblazejewicz** [comment]
@MinaroShikuchi 
```ts
message.PID['PID.5']['PID.5.1']; // $ExpectType string
message.PID['PID.5']['PID.5.1']; // $ExpectType string
```
see also testing section in readme. 
One should just see how that one s expected to be used

**Pull request merged**